### DISCUSSION TITLE
GraphicsView VerticalOptions does not works

### DISCUSSION THREAD

**Discussion by ricardoromaobr** [dotnet/maui#6086] [Q&A] [answered]
I went to trying to put a GraphicsView to fill the full screen on MAUI. 
Here is the code: 
```xaml
<VerticalStackLayout Spacing="25" Padding="30" VerticalOptions="FillAndExpand" BackgroundColor="Yellow">
        <Grid VerticalOptions="FillAndExpand">
        <GraphicsView
            
            Drawable="{StaticResource drawable }"
            
            MoveHoverInteraction="GraphicsView_MoveHoverInteraction"
            StartHoverInteraction="GraphicsView_StartHoverInteraction"
            EndHoverInteraction="GraphicsView_EndHoverInteraction"
            
            VerticalOptions="FillAndExpand" 
            HorizontalOptions="FillAndExpand">
            
            <GraphicsView.GestureRecognizers>
                <TapGestureRecognizer  Tapped="TapGestureRecognizer_Tapped" />
                <PanGestureRecognizer PanUpdated="PanGestureRecognizer_PanUpdated"/>
                <ClickGestureRecognizer Clicked="ClickGestureRecognizer_Clicked" />
            </GraphicsView.GestureRecognizers>
        </GraphicsView>
            
        </Grid>
        <Label x:Name="Status" Text="Teste" VerticalOptions="End"/>
</VerticalStackLayout>
```

Do not its implement yet?

**Comment by danielftz** [comment]
have you tried using just Fill instead of FIllAndExpand?

**Reply by ricardoromaobr** [reply]
I made it to test, but the Fill options behaver like a  wrap content. No not works too. I think that the correct options is FillAndExpand, because if it take content and it not take the complete surface it expand to cover the remain surface.

**Reply by danielftz** [reply]
Im not sure what you mean by wrap content, but FillAndExpand has been deprecated, and it has never worked right in Maui in my experience.

**Reply by ricardoromaobr** [reply]
What are you using to do it. I want just to put GraphicsView and setting it to take full space of the screen. How to do it?

**Reply by davidbritch** [reply]
The *AndExpands layout options don't work in a VerticalStackLayout or a HorizontalStackLayout - they don't exist as concepts in those layouts. If you want to use *AndExpands you have to use them in a StackLayout.

**Comment by hartez** [comment, marked as answer]
The "AndExpand" options are deprecated, but if you absolutely want to use them they will still work with StackLayout (but not VerticalStackLayout or HorizontalStackLayout):

```
<StackLayout Spacing="25" Padding="30" BackgroundColor="Yellow">
		<GraphicsView Drawable="{StaticResource drawable}"
					  VerticalOptions="FillAndExpand" 
					  HorizontalOptions="Fill">
		</GraphicsView>
	<Label x:Name="Status" Text="Test" VerticalOptions="End"/>
</StackLayout>
```

But since you're subdividing a page, I'd recommend using a Grid:

```
<Grid RowDefinitions="*, Auto" RowSpacing="25" Padding="30" BackgroundColor="Yellow">
    <GraphicsView Drawable="{StaticResource drawable}"/>
    <Label x:Name="Status" Text="Test" Grid.Row="1" />
</Grid>
```
